flink-runtime/src/main/java/org/apache/flink/runtime/rest/handler/AbstractRestHandler.java

```diff
@@ -157,7 +160,12 @@ protected void respondAsLeader(final ChannelHandlerContext ctx, Routed routed, T
 						HandlerUtils.sendErrorResponse(ctx, httpRequest, new ErrorResponseBody("Internal server error."), HttpResponseStatus.INTERNAL_SERVER_ERROR);
 					}
 				} else {
-					HandlerUtils.sendResponse(ctx, httpRequest, resp, messageHeaders.getResponseStatusCode());
+					if (resp instanceof WebSocketUpgradeResponseBody) {
+						upgradeToWebSocket(ctx, routed, (WebSocketUpgradeResponseBody) resp);
```

Comment:
help me out here. After the upgrade is complete, which parts of the AbstractRestHandler class are still used?

Reply:
The REST handler is not active after the upgrade is complete, and it would be harmless to remove from the pipeline.  The message handler takes over, reading and writing websocket frames based on typed messages.